homeassistant/components/template/config.py

```diff
@@ -53,6 +59,60 @@
 )
 
 
+async def _async_validate_config_item(hass, config):
+    """Validate template config item."""
+
+    LOGGER.debug("Validating template config: %s", config)
+    blueprint_inputs = None
+    if is_blueprint_instance_config(config):
+        blueprints = async_get_blueprints(hass)
+
+        try:
+            blueprint_inputs = await blueprints.async_inputs_from_config(config)
+        except BlueprintException as err:
+            LOGGER.error(
+                "Failed to generate template from blueprint: %s",
+                err,
+            )
+            raise
+
+        LOGGER.debug("Blueprint inputs: %s", blueprint_inputs.inputs)
+
+        try:
+            config = blueprint_inputs.async_substitute()
+        except UndefinedSubstitution as err:
+            LOGGER.error(
+                "Blueprint '%s' failed to generate template with inputs %s: %s",
+                blueprint_inputs.blueprint.name,
+                blueprint_inputs.inputs,
+                err,
+            )
+            raise HomeAssistantError from err
+
+    config = CONFIG_SECTION_SCHEMA(config)
+    # Add blueprint inputs to entity config
+    if blueprint_inputs:
+        for dom in [
+            BINARY_SENSOR_DOMAIN,
+            BUTTON_DOMAIN,
+            NUMBER_DOMAIN,
+            SELECT_DOMAIN,
+            SENSOR_DOMAIN,
+        ]:
+            if dom in config:
+                for idx in range(len(config[dom])):
+                    config[dom][idx][CONF_BLUEPRINT_INPUTS] = {
```

Comment:
This feels like quite a hacky way to get additional data into the entity configuration, without exposing `blueprint_inputs` into the config schema.

Suggestions?